libosmscout/include/osmscout/DataFile.h

```diff
@@ -144,16 +150,14 @@ namespace osmscout {
   /**
    * Read one data value from the given file offset.
    *
-   * Method is thread-safe.
+   * Method is NOT thread-safe.
    */
   template <class N>
   bool DataFile<N>::ReadData(const TypeConfig& typeConfig,
                              FileScanner& scanner,
                              FileOffset offset,
                              N& data) const
   {
-    std::lock_guard<std::mutex> lock(accessMutex);
```

Comment:
Hi. I added lock to all public methods ( https://github.com/Framstag/libosmscout/pull/232/files#diff-76192701748b5e82dcf359e948e0e36cR435 ) and remove it from private ones. `DataFile` class is still thread-safe.